// services/replicateService.js
// services/replicateService.js - The Demon Forger
import Replicate from 'replicate';

const replicate = new Replicate({
  auth: process.env.REPLICATE_API_TOKEN,
});

export const generateWithReplicate = async (userPrompt) => {
  const finalPrompt = `masterpiece, ultra-premium, photorealistic, insanely detailed, cinematic, dark fantasy, horror, creature design, art by Zdzisław Beksiński and Wayne Barlowe, rendered in Octane, Unreal Engine 5, 8K, ${userPrompt}, towering biomechanical demon forged in the core of a black hole, its body a living volcano of rage and power, asymmetrical body, obsidian armor with gold inlay, biomechanical chitinous plates, fire is a complex multi-layered effect of blinding white-blue plasma, liquid gold, and magma red, soulfire, epic scale`;

  const input = {
    prompt: finalPrompt,
    width: 1024,
    height: 1024,
    refine: "expert_ensemble_refiner",
    scheduler: "K_EULER",
    num_outputs: 1,
    guidance_scale: 7.5,
  };
  
  const model = "stability-ai/sdxl:39ed52f2a78e934b3ba6e2a89f5b1c712de7dfea535525255b1aa35c5565e08b";
  const output = await replicate.run(model, { input });
  return output[0];
};
